src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1491,14 +1492,8 @@ public static <E extends Structure> void registerStructure(Class<E> c, EntryVali
 		structures.add(structureInfo);
 	}
 
-	/**
-	 * Modifications made to the returned Collection will not be reflected in the events available for parsing.
-	 */
 	public static Collection<SkriptEventInfo<?>> getEvents() {
-		// Only used in documentation generation, so generating a new list each time is fine
-		return (Collection<SkriptEventInfo<?>>) (Collection<?>) structures.stream()
-			.filter(info -> info instanceof SkriptEventInfo)
-			.collect(Collectors.toList());
+		return events;
```

Comment:
This is raw access to the events collection. I do not agree with providing mutable collections. Maintain a duplicate collection or make this collection immutable.

Reply:
Every `SyntaxElement` collection getter returns the "raw" collection

Reply:
Yes, but you're breaking existing behavior. Going to label as breaking change.

Reply:
What existing behaviour would be broken with this change?

Reply:
Addons.
An addon could be grabbing or removing syntaxes from this collection, to not realize that it actually didn't apply to the actual collection.

Reply:
What? The events collection is used for parsing events, any modifications done to it will be reflected to the events. I don't understand what your point is

Reply:
@TheLimeGlass

Reply:
Collectors.toList() creates a new list. What else do you need?